### PR TITLE
ARROW-17003: [Java][Docs] Document arrow-jdbc adapter

### PR DESCRIPTION
**Pull request by lidavidm**
Add a basic documentation page for the arrow-jdbc adapter.

I would also like to add a Cookbook page, and then cross-link the two pages, as a follow-up.

### CODE REVIEW

**Comment by lidavidm** [comment]
CC @toddfarmer

**Comment by lidavidm** [comment]
<details>
<summary>Preview</summary>

image

</details>

File: docs/source/java/jdbc.rst [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+   try (ArrowVectorIterator iter = JdbcToArrow.sqlToArrowVectorIterator(rs, config)) {
+     while (iter.hasNext()) {
+       VectorSchemaRoot root = iter.next();
+       // Consume the root…
+     }
+   }
+
+The JDBC type can be explicitly specified, which is useful since JDBC
+drivers can give spurious type information. For example, the Postgres
+driver has been observed to use Decimal types with scale and precision
```
[the remark this answers is not in the document]
Comment (lidavidm) [reply docs/source/java/jdbc.rst:67]: Good point, thanks. I'll clarify the docs when I rebase this on top of #13358

[the remark this answers is not in the document]
Comment (lidavidm) [reply docs/source/java/jdbc.rst:67]: I added a note to the Decimal type in the type mapping table below.

**Review by toddfarmer — APPROVED** [review]

File: docs/source/java/jdbc.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (toddfarmer) [reply docs/source/java/jdbc.rst:67]: Thanks!

**Pull request merged**